pkg/diagnostics/networkpod/pod.go

```diff
@@ -0,0 +1,194 @@
+}
+
+// Description is part of the Diagnostic interface and just returns the diagnostic description.
+func (d CheckPodNetwork) Description() string {
+	return "Check pod to pod communication in the cluster. In case of ovs-subnet network plugin, all pods should be able to communicate with each other and in case of multitenant network plugin, pods in non-global projects should be isolated and pods in global projects should be able to access any pod in the cluster and vice versa."
+}
+
+// CanRun is part of the Diagnostic interface; it determines if the conditions are right to run this diagnostic.
+func (d CheckPodNetwork) CanRun() (bool, error) {
+	if d.KubeClient == nil {
+		return false, errors.New("must have kube client")
+	} else if d.OSClient == nil {
+		return false, errors.New("must have openshift client")
+	}
+	return true, nil
+}
+
+// Check is part of the Diagnostic interface; it runs the actual diagnostic logic
+func (d CheckPodNetwork) Check() types.DiagnosticResult {
+	d.res = types.NewDiagnosticResult(CheckPodNetworkName)
+
+	pluginName, ok, err := util.GetOpenShiftNetworkPlugin(d.OSClient)
+	if err != nil {
+		d.res.Error("DPodNet1001", err, fmt.Sprintf("Checking network plugin failed. Error: %s", err))
+		return d.res
+	}
+	if !ok {
+		d.res.Warn("DPodNet1002", nil, fmt.Sprintf("Skipping pod connectivity test. Reason: Not using openshift network plugin."))
+		return d.res
+	}
+
+	localPods, nonlocalPods, err := util.GetLocalAndNonLocalDiagnosticPods(d.KubeClient)
+	if err != nil {
+		d.res.Error("DPodNet1003", err, fmt.Sprintf("Getting local and nonlocal pods failed. Error: %s", err))
+		return d.res
+	}
+
+	if sdnplugin.IsOpenShiftMultitenantNetworkPlugin(pluginName) {
+		netnsList, err := d.OSClient.NetNamespaces().List(kapi.ListOptions{})
+		if err != nil {
+			d.res.Error("DPodNet1004", err, fmt.Sprintf("Getting all network namespaces failed. Error: %s", err))
+			return d.res
+		}
+
+		d.vnidMap = map[string]uint32{}
+		for _, netns := range netnsList.Items {
+			d.vnidMap[netns.NetName] = netns.NetID
+		}
+	}
+
+	localGlobalPods, localNonGlobalPods := util.GetGlobalAndNonGlobalPods(localPods, d.vnidMap)
+	nonlocalGlobalPods, nonlocalNonGlobalPods := util.GetGlobalAndNonGlobalPods(nonlocalPods, d.vnidMap)
+
+	d.checkSameNodePodToPodConnection(localGlobalPods, localNonGlobalPods)
+	d.checkDifferentNodePodToPodConnection(localGlobalPods, localNonGlobalPods, nonlocalGlobalPods, nonlocalNonGlobalPods)
+	return d.res
+}
+
+func (d CheckPodNetwork) checkDifferentNodePodToPodConnection(localGlobalPods, localNonGlobalPods, nonlocalGlobalPods, nonlocalNonGlobalPods []kapi.Pod) {
+	// Applicable to flat and multitenant networks
+	if len(localGlobalPods) > 0 && len(nonlocalGlobalPods) > 0 {
+		d.checkConnection(localGlobalPods, nonlocalGlobalPods)
+	} else {
+		d.res.Warn("DPodNet1005", nil, fmt.Sprintf("Skipping pod connectivity test for global projects on different nodes. Reason: Couldn't find 2 global pods."))
```

Comment:
Is this really a warn? What should the user do about it, or how should they find out if there is anything to do about it? I can tell you this message isn't likely to mean much to the user as it is. What's a global pod or global project, to a user? Any explanation you can give will reduce support calls.